File: extends/template/domain/template.go
package domain

import "time"

// Template 模版（消息/通知/通用，通过 TemplateType 区分）
type Template struct {
	ID           uint      `json:"id"`
	Name         string    `json:"name"`          // 模板名称
	Code         string    `json:"code"`          // 模板编码（唯一）
	TemplateType string    `json:"template_type"` // general | message | email | sms
	Title        string    `json:"title"`         // 标题模板（支持 {{占位符}}）
	Content      string    `json:"content"`       // 内容模板（支持 {{占位符}}）
	Status       int       `json:"status"`        // 1=启用, 0=禁用
	Remark       string    `json:"remark"`
	Version      string    `json:"version"`
	CreatedAt    time.Time `json:"created_at"`
	UpdatedAt    time.Time `json:"updated_at"`
}

File: java/src/main/java/com/kingfisher/modules/template/domain/Template.java
package com.kingfisher.modules.template.domain;

import com.fasterxml.jackson.annotation.JsonInclude;
import com.fasterxml.jackson.annotation.JsonProperty;
import lombok.Data;

import java.time.LocalDateTime;

/**
 * 模版领域实体，对应 templates 表。
 * 与 Go extends/template/domain.Template 1:1 对齐。
 */
@Data
@JsonInclude(JsonInclude.Include.NON_NULL)
public class Template {
    private Long id;
    private String name;
    private String code;
    @JsonProperty("template_type")
    private String templateType;
    private String title;
    private String content;
    private Integer status;
    private String remark;
    private String version;
    @JsonProperty("created_at")
    private LocalDateTime createdAt;
    @JsonProperty("updated_at")
    private LocalDateTime updatedAt;
}
